### PR TITLE
Beta/split 0.0.0 fixes

### PR DESCRIPTION
**Pull request by deveusss**
- [x] In query_users_balances change "wallet" to "address" to remove confusion
- [x] In instantiate TREASURY_BALANCE starts as zero, but info could have funds - so maybe add funds into TREASURY_BALANCE or just error if info funds attached
- [x] Can not pass funds in factory deploy

### CODE REVIEW

File: contracts/croncat-manager/src/contract.rs [hunk 1]
```diff
@@ -63,6 +63,14 @@ pub fn instantiate(
         .transpose()?
         .unwrap_or_else(|| info.sender.clone());
 
+    let treasury_funds = info
+        .funds
+        .iter()
+        .find(|coin| coin.denom == denom)
+        .ok_or(ContractError::NoTreasuryFundsAttached {})?;
```
Comment (Buckram123) [inline contracts/croncat-manager/src/contract.rs:70]: Let's say we have this situation: 
- [20coinA, 30coinB] attached and `config.denom` is `coinA`

Should we fail tx if other coins were attached or what to do with coinB? @TrevorJTClarke

Comment (Buckram123) [inline contracts/croncat-manager/src/contract.rs:70]:
```suggestion
```
It should allow empty deposit!

File: contracts/croncat-manager/src/tests/contract_tests.rs [hunk 2]
```diff
@@ -361,7 +369,9 @@ fn cw20_bad_messages() {
     let mut app = default_app();
 
     let instantiate_msg: InstantiateMsg = default_instantiate_message();
-    let manager_addr = init_manager(&mut app, instantiate_msg, &[]).unwrap();
+    let send_funds: &[Coin] = &[coin(600, instantiate_msg.denom.clone())];
+
+    let manager_addr = init_manager(&mut app, instantiate_msg, send_funds).unwrap();
```
Comment (Buckram123) [inline contracts/croncat-manager/src/tests/contract_tests.rs:374]: Let's not have it standard way to initialize contract

**Review by Buckram123 — CHANGES_REQUESTED** [review]
I personally against attached deposit being the standard way to initialize treasury balance, it doesn't make sense if croncat-factory owner is a croncat-dao itself, why would they ping-pong balance in and out?

Can't think it can be useful for anything, so I'm more towards blocking attached funds on initialize, at least for now! What's your thoughts on that? @TrevorJTClarke  @mikedotexe @NinoLipartiia @deveusss

File: contracts/croncat-manager/src/contract.rs (hunk 1 above)
Comment (deveusss) [reply contracts/croncat-manager/src/contract.rs:70]: Trevors comment "TREASURY_BALANCE or just error if no funds attached"

File: contracts/croncat-manager/src/tests/contract_tests.rs (hunk 2 above)
Comment (deveusss) [reply contracts/croncat-manager/src/tests/contract_tests.rs:374]: Copy it from agent tests

File: contracts/croncat-manager/src/contract.rs (hunk 1 above)
Comment (deveusss) [reply contracts/croncat-manager/src/contract.rs:70]: For now its not failing if others are attached by if no funds with default denom - it will fail

Comment (Buckram123) [reply contracts/croncat-manager/src/contract.rs:70]: "or just error if info funds attached"
pretty sure it means
```rust
if info.funds.is_empty{
    return Err();
}
```

File: contracts/croncat-manager/src/tests/contract_tests.rs (hunk 2 above)
Comment (Buckram123) [reply contracts/croncat-manager/src/tests/contract_tests.rs:374]: I mean about positive balance at treasury from the start)

File: contracts/croncat-manager/src/contract.rs (hunk 1 above)
Comment (deveusss) [reply contracts/croncat-manager/src/contract.rs:70]: ahh, true

Comment (deveusss) [reply contracts/croncat-manager/src/contract.rs]: updated it, so now funds are added into treasure balance

File: contracts/croncat-manager/src/balances.rs [hunk 3]
```diff
@@ -230,7 +230,7 @@ pub fn execute_refill_native_balance(
         .may_load(deps.storage, task_hash.as_bytes())?
         .ok_or(ContractError::NoTaskHash {})?;
 
-    if info.funds.len() > 2 {
+    if info.funds.len() > 3 {
```
Comment (TrevorJTClarke) [inline contracts/croncat-manager/src/balances.rs:233]: @deveusss @Buckram123 whats the hardcoded value here for? Is there a reason its not higher like 10 or just simply 1?

File: justfile [hunk 4]
```diff
@@ -14,8 +14,8 @@ build:
 	cargo build --release --lib --target wasm32-unknown-unknown
 deploy:
 	#!/bin/bash
-	chmod +x ./scripts/testnet/deploy.sh
-	./scripts/testnet/deploy.sh -w # only wasm update
+	cd ./scripts/deployment
+	yarn go
```
Comment (TrevorJTClarke) [inline justfile:18]: @deveusss whats the core use-case here? Will CI/CD be running this? If so, can we guarantee other things like build & optimize run first?

File: scripts/deployment/src/manager.ts [hunk 5]
```diff
@@ -20,6 +20,7 @@ export class ManagerClient {
 		const codeId = uploadRes.codeId
 		const checksums = await getChecksums()
 		const githash = await getGitHash()
+		const funds:Coin[]=[coin(50,denom)];
```
Comment (TrevorJTClarke) [inline scripts/deployment/src/manager.ts:23]: @deveusss likely need some `cargo fmt --` love here

**Review by TrevorJTClarke — APPROVED** [review]
@deveusss lookin good! couple minor things.

RE: Things @Buckram123 brought up - theres some back in forth in my head about funds for future needs, i would rather support this way to start and remove if need be.

File: contracts/croncat-manager/src/balances.rs (hunk 3 above)
Comment (Buckram123) [reply contracts/croncat-manager/src/balances.rs:233]: Didn't we decide to keep up to 1 `config.denom` and 1 non-`config.denom`(ibc) same with up to 1 cw20?

Comment (Buckram123) [reply contracts/croncat-manager/src/balances.rs:233]: It should be here then, or limitless if you think it's a bad idea

**Comment by deveusss** [comment]
> @deveusss lookin good! couple minor things.
> 
> RE: Things @Buckram123 brought up - theres some back in forth in my head about funds for future needs, i would rather support this way to start and remove if need be.

So we are supporting passing funds in instantiate?

**Comment by deveusss** [comment]
> I personally against attached deposit being the standard way to initialize treasury balance, it doesn't make sense if croncat-factory owner is a croncat-dao itself, why would they ping-pong balance in and out?
> 
> Can't think it can be useful for anything, so I'm more towards blocking attached funds on initialize, at least for now! What's your thoughts on that? @TrevorJTClarke @mikedotexe @NinoLipartiia @deveusss

Whats wrong with passing funds into your contract at init? why not if we have this possibility?

File: justfile (hunk 4 above)
Comment (deveusss) [reply justfile:18]: this is not for CI/CD but for deploy on testnet and check things in testnet

File: scripts/deployment/src/manager.ts (hunk 5 above)
Comment (deveusss) [reply scripts/deployment/src/manager.ts]: √

**Review by Buckram123 — APPROVED** [review]

File: contracts/croncat-agents/src/contract.rs [hunk 6]
```diff
@@ -176,7 +173,10 @@ fn query_get_agent_tasks(
 
     let (block_slots, cron_slots) = croncat_tasks_contract::query_tasks_slots(deps, &config)?;
     if block_slots == 0 && cron_slots == 0 {
-        return Ok(None);
+        return Ok(AgentTaskResponse {
+            num_cron_tasks: Uint64::zero(),
+            num_block_tasks: Uint64::zero(),
+        });
```
[the remark this answers is not in the document]
Comment (deveusss) [reply contracts/croncat-agents/src/contract.rs:179]: this is old code migrated form old contract)))

File: contracts/croncat-agents/src/contract.rs [hunk 7]
```diff
@@ -410,7 +410,7 @@ fn unregister_agent(
     let mut active_agents: Vec<Addr> = AGENTS_ACTIVE.load(deps.storage)?;
     if let Some(index) = active_agents.iter().position(|addr| addr == agent_id) {
         //Notify the balancer agent has been removed, to rebalance itself
-        AGENT_TASK_DISTRIBUTOR.on_agent_unregistered(deps.storage, agent_id)?;
+        AGENT_STATS.remove(deps.storage, agent_id);
```
Comment (TrevorJTClarke) [inline contracts/croncat-agents/src/contract.rs:413]: Thank you!

File: contracts/croncat-agents/src/distributor.rs [hunk 8]
```diff
@@ -104,33 +101,20 @@ impl<'a> RoundRobinAgentTaskDistributor<'a> for AgentTaskDistributor {
                 }
                 let agent_tasks_total = total_tasks.saturating_div(agent_count) + extra;
 
-                Ok((agent_tasks_total.into(), extra.into()))
+                Ok(agent_tasks_total.into())
```
Comment (TrevorJTClarke) [inline contracts/croncat-agents/src/distributor.rs:104]: So much better!!

**Review by TrevorJTClarke — APPROVED** [review]

**Pull request merged**